local.go

```diff
@@ -325,20 +355,18 @@ func checkConvergence(loc *Location, iterType IterationType, stats *Stats, setti
 		}
 	}
 
-	if iterType == MajorIteration && settings.MajorIterations > 0 {
-		if stats.MajorIterations >= settings.MajorIterations {
-			return IterationLimit
-		}
-	}
 	return NotTerminated
 }
 
-// invalidate marks all fields of Location with NaNs and it is the
-// responsibility of Method to issue such EvaluationTypes that make Location
-// valid again. Invalidation is done internally as a help for implementers to
-// detect silent bugs in Methods using inconsistent Location, e.g., using
-// Gradient without a GradEvaluation request.
-func invalidate(loc *Location) {
+// evaluate evaluates the problem p at loc.X using the evaluation eval,
```

Comment:
evaluate evaluates the routines specified by the Operation at loc.X, storing the answer into loc and updating stats.